pubnub/endpoints/endpoint.py

```diff
@@ -99,7 +100,8 @@ def request_headers(self):
             headers["Content-Encoding"] = "gzip"
         if self.http_method() == HttpMethod.POST:
             headers["Content-type"] = "application/json"
-
+        if self._custom_headers:
```

Comment:
https://peps.python.org/pep-0008/#programming-recommendations
Initially `self._custom_headers` is equal `None`. In this case you should check if your variable is None or not. I assume that `custom_headers` variable is a dict. So you can change initial value from None to empty dict ;)